### PR TITLE
Adds support for propshaft (#2)

### PR DESCRIPTION
**Pull request by jvillarejo**
I am migrating my webapp to use propshaft and dartsass and removing sprockets as the asset pipeline.

This PR introduce a new `AssetPropshaftProvider` to resolve the assets instead of the `AssetPipelineProvider`.

Researching `propshaft` more I ended replicating a similar idea implemented in the `Propshaft::Server` class.

We need to provide `roadie` with the compiled stylesheet, so it can:
 - Reference the correct digested assets when using `url(...)` on the stylesheet.
 - Compile the whole stylesheet if `sass` is being used.  

Now the integration_spec works without requiring the assets to be compiled. 
It serves the compiled assets when running in development environment (which the dummy apps run) 

I also added expectations for the body ``background-color: green` inlining.

It fixes https://github.com/Mange/roadie-rails/issues/112

### CODE REVIEW

**Comment by PikachuEXE** [comment]
CI failed
https://github.com/Mange/roadie-rails/actions/runs/9022266978/job/24797354807?pr=114

Please fix when you can~

**Comment by jvillarejo** [comment]
There are some files that appear on the linter when running `bundle exec standardrb` that I didn't change
This ones:

```
  lib/roadie/rails/mailer.rb:11:9: Style/ArgumentsForwarding: Use anonymous block arguments forwarding (`&`).
  lib/roadie/rails/mailer.rb:13:31: Style/ArgumentsForwarding: Use anonymous block arguments forwarding (`&`).
  lib/roadie/rails/options.rb:109:23: Performance/StringIdentifierArgument: Use `:"#{option}="` instead of `"#{option}="`.
```

¿Should I fix them too on this PR ?

**Comment by PikachuEXE** [comment]
Yes please

**Comment by jvillarejo** [comment]
There is a problem here. `standardrb` mark as linter errors this ones:

```ruby
lib/roadie/rails/mailer.rb:11:9: Style/ArgumentsForwarding: Use anonymous block arguments forwarding (`&`).
lib/roadie/rails/mailer.rb:13:31: Style/ArgumentsForwarding: Use anonymous block arguments forwarding (`&`).
lib/roadie/rails/options.rb:109:23: Performance/StringIdentifierArgument: Use `:"#{option}="` instead of `"#{option}="`.
```

Fixing the one in `lib/roadie/rails/options.rb` shouldn't present any problem. 

But the anonymous block arguments forwarding is a feature added in ruby 3.1
If I change that, the library won't be compatible with previous ruby versions. 

I don't think that's part of the PR. but I don't know why the linter is now failing.  

What should we do?
I can [ignore the errors with a comment.](https://github.com/standardrb/standard?tab=readme-ov-file#ignoring-a-line-with-a-comment)

**Comment by PikachuEXE** [comment]
Coz the lint job is run on ruby 3.2
Please ignore with config file
https://github.com/standardrb/standard#ignoring-specific-rules-in-files-and-globs
Better have one place to track all ignore rules (will be removed when support for earlier ruby versions dropped)

File: .standard.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ignore:
+  - 'lib/roadie/rails/mailer.rb':
+    - Style/ArgumentsForwarding 
+  - 'lib/roadie/rails/options.rb':
+    - Performance/StringIdentifierArgument
```
Comment (PikachuEXE) [inline .standard.yml:5]: Missing new line at the end again :P
I will add https://editorconfig.org and set `insert_final_newline`
You should setup your editor(s)/IDE(s) to follow https://editorconfig.org config if present (though I have never seen any project allowing missing new line
I will simply merge and fix it later

Comment (jvillarejo) [reply .standard.yml:5]: Woops sorry!

**Comment by PikachuEXE** [comment]
CI failing now due to actual path shorten
I have no idea if it's the default config changed or the setup in this project's testing causing it...
https://github.com/Mange/roadie-rails/actions/runs/9326040234

**Comment by jvillarejo** [comment]
I will check it out.

**Comment by PikachuEXE** [comment]
@jvillarejo Any progress?

**Comment by jvillarejo** [comment]
Sorry @PikachuEXE I been having some rough weeks. I had this on my TODO list.  
I will try to do fix it this week. 
Thanks for the link !

**Comment by PikachuEXE** [comment]
It's simply test case no urgency at all! :)

**Pull request merged**